test/modules/encoders/test_swin_transformer_3d_encoder.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+
+import torch
+from torchmultimodal.modules.encoders.swin_transformer_3d_encoder import (
+    PatchEmbed3d,
+    PatchMerging3d,
+    ShiftedWindowAttention3d,
+    SwinTransformer3dEncoder,
+)
+from torchmultimodal.utils.common import get_current_device
+
+
+class TestSwinTransformer3dEncoder(unittest.TestCase):
+    def setUp(self):
+        torch.manual_seed(42)
+        self.device = get_current_device()
+
+        # Setup Encoder to test
+        self.encoder = (
+            SwinTransformer3dEncoder(
+                patch_size=(2, 4, 4),
+                embed_dim=96,
+                depths=[2, 2, 6, 2],
+                num_heads=[3, 6, 12, 24],
+                window_size=(8, 7, 7),
+                stochastic_depth_prob=0.2,
+                norm_layer=torch.nn.LayerNorm,
+                patch_embed=PatchEmbed3d,
+            )
+            .to(self.device)
+            .eval()
+        )
+
+    def test_swin_transformer_3d_encoder(self):
+        torch.manual_seed(42)
```

Comment:
redundant? The seed fixed in `setUp` should prevail through the methods right?

Reply:
Oh, I didn't know that.... I assume it is only called once like __init__ because the value that I have before I add new function dont work anymore on my laptop and I need to change it. Let me double verify this behaviour.

Reply:
@langong347  you are right. I have confirm that setUp indeed called on every method. Hmm... then the problem I encountered before might caused by other thing. Anyway I have added the seed init on setUp instead now.